stl/inc/algorithm

```diff
@@ -1046,13 +1072,39 @@ namespace ranges {
                     --_Final2; // since ranges have equal lengths, _Final2 cannot equal _First2
 
                     if (!_STD invoke(_Pred, _STD invoke(_Proj1, *_Final1), _STD invoke(_Proj2, *_Final2))) { // mismatch
-                        ++_Final1;
-                        ++_Final2;
                         break;
                     }
                 }
                 // If we get here, initial elements do not match, final elements do not match, and ranges have length
                 // at least 2.
+
+                const auto _ProjectedPred = [&]<class _Ty1, class _Ty2>(_Ty1&& _Left, _Ty2&& _Right) {
+                    return _STD invoke(_Pred, _STD invoke(_Proj1, _STD forward<_Ty1>(_Left)),
+                        _STD invoke(_Proj2, _STD forward<_Ty2>(_Right)));
+                };
+
+                for (;;) {
+                    _TrimResult _Res = _Trim_reversed(_First1, _Final1, _First2, _Final2, _ProjectedPred);
+                    if (_Res != _TrimResult::_KeepTrimming) {
+                        if (_Res == _TrimResult::_Break) {
+                            break;
+                        }
+                        return _Res == _TrimResult::_ReturnTrue;
+                    }
+
+                    _Res = _Trim_equal(_First1, _Final1, _First2, _Final2, _ProjectedPred);
+                    if (_Res != _TrimResult::_KeepTrimming) {
+                        if (_Res == _TrimResult::_Break) {
+                            break;
+                        }
+                        return _Res == _TrimResult::_ReturnTrue;
+                    }
+                }
+
+                ++_Final1;
+                ++_Final2;
+                // If we get here, initial elements do not match, final elements do not match, and ranges have length
+                // at least 2.

```

Comment:
Ditto nitpick as above.

Reply:
Added the same comments. (As there are only 2 occurrences, I felt it was simpler to just repeat the comments instead of saying "look over there", which is slightly hard to maintain over time.)